src/WebJobs.Script/OutOfProc/Rpc/RpcInitializationService.cs

```diff
@@ -78,6 +81,7 @@ public async Task StartAsync(CancellationToken cancellationToken)
         public async Task StopAsync(CancellationToken cancellationToken)
         {
             _logger.LogDebug("Shuttingdown Rpc Channels Manager");
+            await DrainBuffersWithTimeout(_bufferDrainTimeoutInSeconds);
```

Comment:
since functionDispatcher has all the info about invocations, we can handle this just in FunctionDispatcher. Lets discuss offline